content/articles/oop-in-java-abstraction-and-encapsulation/index.md

```diff
@@ -1,350 +1,482 @@
+
+  }
 
-       System.out.println("The dog name is:  " + littleDog.name);
-   }
 }
+
 ```
 
 Now you see, we’ve created a new dog with the name Bingo, you can go on to create as many dogs with different states or attributes.
 
-
 Here is the output
 
 Dog running
+
 Is the dog running:? true
 
 Sleeping
+
 is the dog sleeping:? true
 
 Dog barking...
+
 Is the dog barking:? true
 
 Eating
+
 Is the dog eating:? true
 
 The dog name is: Bingo
 
 Now that we’ve got this out of the way, let's talk about Abstraction and Encapsulation properly.
 
-#**ABSTRACTION**
+# ABSTRACTION
+
 Who could have imagined that Abstraction will be a thing in programming? Who? But right now you can’t talk about OOP without mentioning Abstraction. It is a major building block of OOP. So then, what is Abstraction? In this section, I will explain what Abstraction is and how to use it as a Java developer.
 
-In Abstraction, only the useful attributes of an object are on display or accessible while the unnecessary details are “hidden” or abstracted. For example, while you’re typing on your computer keyboard you’re only interested in punching the keys and believing that it produces results on your screen but you know nothing about the mechanism that produces the result and I’m sure you haven’t even thought about it. Have you? Another example to make things clearer,  imagine a car driver, he is only interested in starting, moving, and stopping the car. Accelerate or change gear and brake but he is not interested in the mechanism that works internally to perform all the above functions. Simply speaking, in abstraction the implementation details are hidden from the user while the functionality is exposed to the user hence the user will only know what a function(method in java) does but not how it does it.
+In Abstraction, only the useful attributes of an object are on display or accessible. The unnecessary details are “hidden”.
+For example, while you’re typing on your computer. You’re only interested in punching the keys believing that it produces results on your screen but you know nothing about the mechanism that produces the result and I’m sure you haven’t even thought about it yet. Have you? 
+Another example to make things clearer,  imagine a car driver, he is only interested in starting, moving, and stopping the car. Accelerate or change gear and brake but he is not interested in the mechanism that works internally to perform all the above functions. Simply, in abstraction, the implementation details are hidden from the user while the functionality is exposed to the user. 
+The while the user will only know what a function(method in java) does but not how it does it.
 
 How to implement Abstraction in Java
+
 In Java, abstraction is implemented in either of these two ways
--[1]: by using the non-access modifier ‘abstract’ keyword. 
--[2]: Interface
 
+- By using the non-access modifier ‘abstract’ keyword. 
```

Comment:
> - By using the non-access modifier ‘abstract’ keyword. 
> 
> - Interface
> 
> - Non-Access Modifier keyword “Abstract” Implementation

This sounds a little bit awkward. You need to phrase these points as though they are a part of the same sentence as the clause before it (i.e. *In Java, abstraction is implemented in either of these two ways*). What I mean by this is that when you read: 

> In Java, abstraction is implemented in either of these two ways
> 
> - By using the non-access modifier ‘abstract’ keyword.
> - Interface
> - Non-Access Modifier keyword “Abstract” Implementation

it should read like one well-flowing sentence.